tgui/packages/tgui/interfaces/PumpControl.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Romayne (https://github.com/MeggalBozale)
+ * @license ISC (https://choosealicense.com/licenses/isc/)
+ */
+
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Section, Slider, Stack } from '../components';
+// import { DataInputOptions } from './common/DataInput';
+import { Window } from '../layouts';
+
+// List of information about a pump
+type PumpData = {
+  tag: string; // Pump name
+  netid: string; // Pump id
+  power: string; // On or off
+  target_output: number; // Current output target of the pump
+  min_output: number;
+  max_output: number;
+  area_name: string; // Name of the area this pump is in
+  processing: boolean; // Whether we are waiting for packet response or not
+  alive: number; // A value of -1, 0, or 1 where -1 is checking if the pump is alive, 0 is dead, and 1 is alive
+}
+
+// List of areas which have pumps
+type AreaList = {
+  area_list: { [key: string] : { [key: string] : PumpData } };
+  frequency: number;
+};
```

Comment:
Same as with the TTV UI, these should be in a separate file

Reply:
moved to /PumpControl